sim-lib/src/defined_activity.rs

```diff
@@ -37,7 +39,11 @@ impl DestinationGenerator for DefinedPaymentActivity {
 
 impl PaymentGenerator for DefinedPaymentActivity {
     fn next_payment_wait(&self) -> Duration {
-        self.wait
+        let wait = Duration::from_secs(self.wait.value() as u64);
```

Comment:
This is the only bit I'm not super happy about, given we'll be building a `Duration` object every time we call `next_payment_wait`, even for wait payments. However, changing this would imply changing the generic for ValueOrRange, and I think potentially more code bolilerplate.

I don't think this is too bad, but I'm open to change it if you find it is.

Reply:
We could have a `RangePaymentActivity` impl of `PaymentGenerator` which deals exclusively with the range case?

We can stil parse everything as-is here, but then create either `RangePaymentActivity` or `DefinedPaymentActivity` when we [switch over from](https://github.com/bitcoin-dev-project/sim-ln/blob/main/sim-cli/src/main.rs#L185) `ActiviyParser` -> `ActivityDefinition`. Might cause a little bit of trait object agony in `main`, but we _should_  be mostly okay in `lib` 🤞 

That might also be quite boiler-platey, but could be worth considering!

Reply:
I don't know if that would be even more complex.

We have two fields that can be either a range or a value (but only one of them has the initialization issue), how do we deal with the combination of these?

`DefinedPaymentActivity` would need to have `wait: Duration, amount: ValueOrRange<u64>` while `RangePaymentActivity` would have `wait: ValueOrRange<u16>, amount: ValueOrRange<u64>`. That's really counterintuitive. The alternative would be to only allow either both or none to be ranges, but that's limiting the functionality.

None of the solutions I've been trying to come up with have been satisfying for one reason or another (nor even having a two-argument generic, since deserialization becomes a mess). I've been checking the constructor of `Duration` and, honestly, it doesn't look that bad (specially considering that we are constructing with `nanos=0`): https://doc.rust-lang.org/src/core/time.rs.html#199-207

So I would bite the bullet with this one, but I'm happy to be convinced of an alternative solution if you could find something I'm missing

Reply:
Managed to completely miss this comment - sorry!! 

> DefinedPaymentActivity would need to have wait: Duration, amount: ValueOrRange<u64> while RangePaymentActivity would have wait: ValueOrRange<u16>, amount: ValueOrRange<u64>. That's really counterintuitive. The alternative would be to only allow either both or none to be ranges, but that's limiting the functionality.

Good point, didn't think about the inbetween case where we only want a range for one of them! Agreed, let's just go ahead w/ what we have here.

Reply:
np, I'll update it with the most recent comments